sdks/python/apache_beam/transforms/trigger.py

```diff
@@ -1064,6 +1064,17 @@ def __init__(self, defensive_copy=True):
     self.global_state = {}
     self.defensive_copy = defensive_copy
 
+  def custom_copy(self):
+    cloned_object = copy.copy(self)
+    cloned_object.timers = copy.deepcopy(self.timers)
+    cloned_object.global_state = copy.deepcopy(self.global_state)
```

Comment:
<!--new_thread; commit:8650ea65963f8e3efc5555d8b3aa5474292efd20; resolved:0-->
We should do similar logic to the inner loop below here too.